clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2250,64 +2293,77 @@ namespace {
       // - bounds(any), there is no runtime checks.
       // - bounds(lb, ub):  If the declared bounds of the cast operation are
       // (e2, e3),  a runtime check that lb <= e2 && e3 <= ub is inserted
-      // during code generation.
-      if (CK == CK_DynamicPtrBounds) {
-        Expr *SubExpr = E->getSubExpr();
-
-        CHKCBindTemporaryExpr *TempExpr = dyn_cast<CHKCBindTemporaryExpr>(SubExpr);
-        assert(TempExpr);
-
-        // These bounds will be computed and tested at runtime.  Don't
-        // recompute any expressions computed to temporaries already.
-        Expr *TempUse = CreateTemporaryUse(TempExpr);
-
-        Expr *SubExprAtNewType = CreateExplicitCast(E->getType(), CastKind::CK_BitCast, 
-                                                    TempUse, true);
-        BoundsExpr *DeclaredBounds = E->getBoundsExpr();
-        BoundsExpr *NormalizedBounds = ExpandToRange(SubExprAtNewType,
-                                                       DeclaredBounds);
-        BoundsExpr *SubExprBounds = InferRValueBounds(TempUse, CSS, Facts);
-        if (SubExprBounds->isUnknown()) {
-          S.Diag(SubExpr->getBeginLoc(), diag::err_expected_bounds);
-        }
-
-        assert(NormalizedBounds);
-
-        E->setNormalizedBoundsExpr(NormalizedBounds);
-        E->setSubExprBoundsExpr(SubExprBounds);
-        if (DumpBounds)
-          DumpBoundsCastBounds(llvm::outs(), E, DeclaredBounds,
-                               NormalizedBounds, SubExprBounds);
+      // during code generation.
+      if (CK == CK_DynamicPtrBounds || CK == CK_AssumePtrBounds) {
+        CHKCBindTemporaryExpr *TempExpr = dyn_cast<CHKCBindTemporaryExpr>(SubExpr);
+        assert(TempExpr);
+
+        // These bounds will be computed and tested at runtime.  Don't

```

Comment:
I think this should say "These bounds may be computed and tested at runtime."   Tthe original comment in the code was flawed.  The flaw is more obvious here.